### PR TITLE
Allow multiple custom locales files

### PR DESCRIPTION
**Pull request by AdrienClairembault**
GLPI allow custom locales in `files/_locales`.
Locale for core are placed in `files/_locales/core` and for a given plugin in `files/_locales/given_plugin_name`.

The main issue with this system is that it only allows one folder per locale "namespace".

To improve this I've modified this behavior to look for `files/_locales/core`  AND `files/_locales/core_*`  files.
This means that two  `core_feature1` and a `core_feature2` folders can coexist and add their own locales without having to be merged in one folder.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: inc/plugin.class.php [hunk 1]
```diff
@@ -360,17 +360,32 @@ static function loadLang($plugin_key, $forcelang = '', $coretrytoload = '') {
...
-         $TRANSLATE->addTranslationFile('phparray', $phpfile, $plugin_key, $coretrytoload);
-      }
+         if ($dir == $plugin_key) {
+            return true;
+         }
 
-      // Load local MO file if it exists -- keep last so it gets precedence
-      if (file_exists($mofile)) {
-         $TRANSLATE->addTranslationFile('gettext', $mofile, $plugin_key, $coretrytoload);
+         return strpos($dir, $plugin_key . '_') !== false;
```
Comment (cedric-anne) [inline inc/plugin.class.php:373]: Using a regex would be more secure.
For instance, when loading `dashboard` plugin locales, we do not want to load `mydashboard` and `advanceddashboard` custom locales.
```php
return preg_match('/^' . preg_quote($plugin_key, '/') . '(_.+)?$'/', $dir);
```

Comment (cedric-anne) [inline inc/plugin.class.php:373]: Using a regex would be more secure.
For instance, when loading `dashboard` plugin locales, we do not want to load `mydashboard` and `advanceddashboard` custom locales.
```php
return preg_match('/^' . preg_quote($plugin_key, '/') . '(_.+)?$'/', $dir);
```

File: inc/session.class.php [hunk 2]
```diff
@@ -637,17 +637,32 @@ static function loadLanguage($forcelang = '', $with_plugins = true) {
...
-         $TRANSLATE->addTranslationFile('phparray', $phpfile, 'glpi', $trytoload);
-      }
+         if ($dir == 'core') {
+            return true;
+         }
 
-      // Load local MO file if it exists -- keep last so it gets precedence
-      if (file_exists($mofile)) {
-         $TRANSLATE->addTranslationFile('gettext', $mofile, 'glpi', $trytoload);
+         return strpos($dir, 'core_') !== false;
```
Comment (cedric-anne) [inline inc/session.class.php:650]:
```php
return preg_match('/^core(_.+)?$'/', $dir);
```

Comment (trasher) [reply inc/session.class.php:650]: ```Toolbox::startsWith()`` would do the same, without using a regexp.

File: inc/plugin.class.php (hunk 1 above)
Comment (AdrienClairembault) [reply inc/plugin.class.php:373]: Good catch, I wanted to check strpos === 0 to prevent that but forgot. Guess `Toolbox::startsWith` will do fine too :)

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**